contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,349 @@
+        require(
+            msg.sender == validatorRegistrator,
+            "Caller is not the Registrator"
+        );
+        _;
+    }
+
+    /// @dev Throws if called by any account other than the Staking monitor
+    modifier onlyStakingMonitor() {
+        require(msg.sender == stakingMonitor, "Caller is not the Monitor");
+        _;
+    }
+
+    /// @dev Throws if called by any account other than the Strategist
+    modifier onlyStrategist() {
+        require(
+            msg.sender == IVault(VAULT_ADDRESS).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        _;
+    }
+
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _vaultAddress Address of the Vault
+    /// @param _beaconChainDepositContract Address of the beacon chain deposit contract
+    /// @param _ssvNetwork Address of the SSV Network contract
+    /// @param _maxValidators Maximum number of validators that can be registered in the strategy
+    constructor(
+        address _wethAddress,
+        address _vaultAddress,
+        address _beaconChainDepositContract,
+        address _ssvNetwork,
+        uint256 _maxValidators
+    ) {
+        WETH = _wethAddress;
+        BEACON_CHAIN_DEPOSIT_CONTRACT = _beaconChainDepositContract;
+        SSV_NETWORK = _ssvNetwork;
+        VAULT_ADDRESS = _vaultAddress;
+        MAX_VALIDATORS = _maxValidators;
+    }
+
+    /// @notice Set the address of the registrator which can register, exit and remove validators
+    function setRegistrator(address _address) external onlyGovernor {
+        validatorRegistrator = _address;
+        emit RegistratorChanged(_address);
+    }
+
+    /// @notice Set the address of the staking monitor that is allowed to reset stakeETHTally
+    function setStakingMonitor(address _address) external onlyGovernor {
+        stakingMonitor = _address;
+        emit StakingMonitorChanged(_address);
+    }
+
+    /// @notice Set the amount of ETH that can be staked before staking monitor
+    // needs to a approve further staking by resetting the stake ETH tally
+    function setStakeETHThreshold(uint256 _amount) external onlyGovernor {
+        stakeETHThreshold = _amount;
+        emit StakeETHThresholdChanged(_amount);
+    }
+
+    /// @notice Reset the stakeETHTally
+    function resetStakeETHTally() external onlyStakingMonitor {
+        stakeETHTally = 0;
+        emit StakeETHTallyReset();
+    }
+
+    /// @notice Stakes WETH to the node validators
+    /// @param validators A list of validator data needed to stake.
+    /// The `ValidatorStakeData` struct contains the pubkey, signature and depositDataRoot.
+    /// Only the registrator can call this function.
+    // slither-disable-start reentrancy-eth
+    function stakeEth(ValidatorStakeData[] calldata validators)
+        external
+        onlyRegistrator
+        whenNotPaused
+    {
+        uint256 requiredETH = validators.length * 32 ether;
```

Comment:
Should this function switch to using FULL_STAKE instead of 32 ether? Would mean we move the definition of FULL_STAKE down into this contract

Reply:
Yes, I've switched it and a few other spots in `ValidatorRegistrator` to use FULL_STAKE instead of 32 ether. It puts us in a better place for EIP-7251, Increase the MAX_EFFECTIVE_BALANCE